src/p2p/net_node.inl

```diff
@@ -2292,7 +2314,7 @@ namespace nodetool
     CHECK_AND_ASSERT_MES(address_ok, false,
         "Only IPv4 or IPv6 addresses are supported here");
 
-    const epee::net_utils::network_address na = context.m_remote_address;
+    const epee::net_utils::network_address na = get_preferred_address(context, node_data);
```

Comment:
The code below is now busted, the cast to ipv6 can fail if a non IPv4 or IPV6 address is given. See other comment about ensuring the addresses are from the same zone.